app/src/main/java/com/github/factotum_sdp/factotum/models/AddressCoordinates.kt

```diff
@@ -0,0 +1,170 @@
+            store(this, context)
+        }
+    }
+
+
+    companion object {
+        const val CACHE_FILE_NAME = "locations.txt"
+        const val CACHE_FILE_SEPARATOR = "|"
+        const val MAX_RESULT = 4
+
+        /**
+         * Blocking function that returns a list of addresses dependant on the query
+         *
+         * @param query : String. Address that we want to search
+         * @param context : Context. Context in which this function is called
+         * @return : List<Location>?. Null if no result
+         */
+        fun geocoderQuery(query: String, context: Context): List<AddressCoordinates>? {
+            // must handle differently depending on SDK.
+            // getLocationFromName(String, int) deprecated in SDK 33
+            if(query.length < 2) return null
+            val geocoder = Geocoder(context)
+            if (!isNetworkAvailable(context) || !Geocoder.isPresent()) {
+                return searchInCache(query, context)
+            }
+            return if (Build.VERSION.SDK_INT >= TIRAMISU) {
+                tiramisuResultHandler(query, geocoder, context)
+            } else {
+                Log.e("Location", "SDK < TIRAMISU")
+                return null
+                //resultHandler(query, geocoder, context)
+            }
+        }
+
+        fun searchInCache(query: String, context: Context): List<AddressCoordinates>?{
+            val cacheFile = File(context.cacheDir, CACHE_FILE_NAME)
+            var addressCoordinates : List<AddressCoordinates>? = emptyList()
+            cacheFile.forEachLine { line ->
+                if (line.contains(query)) {
+                    val split = line.split(CACHE_FILE_SEPARATOR)
+                    val result = AddressCoordinates(split[0], LatLng(split[1].toDouble(), split[2].toDouble()), context)
+                    addressCoordinates = addressCoordinates?.plus(result)
+                }
+            }
+            return addressCoordinates
+        }
+
+        @RequiresApi(TIRAMISU)
+        private fun tiramisuResultHandler(query: String, geocoder: Geocoder, context: Context): List<AddressCoordinates>? {
+            val result = CompletableFuture<List<AddressCoordinates>?>()
+            geocoder.getFromLocationName(query, MAX_RESULT) { addresses ->
+                val queryResults = if (addresses.size > 0) addresses.map{
+                    AddressCoordinates(it.getAddressLine(0), LatLng(it.latitude, it.longitude), context, false)
+                } else null
+                result.complete(queryResults)
+            }
+            val listToReturn : List<AddressCoordinates>?
+            try {
+                listToReturn = result.get(2L, java.util.concurrent.TimeUnit.SECONDS)
+            } catch (e: Exception) {
+                Log.e("Location", "Error getting location from geocoder",e)
+                return null
+            }
+            return listToReturn
+        }
+/* For now do not need backward handling anymore
```

Comment:
Large commented code part which should not be merged on the main branch.
If the reason was to don't lose that code, it is not problem because it is kept in the git history.